core/src/main/java/org/elasticsearch/ingest/core/CompoundProcessor.java

```diff
@@ -16,7 +16,7 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-
+//TODO(simonw): can all these classes go into org.elasticsearch.ingest?
```

Comment:
the idea would be to keep two packages as the .core package should be es agnostic, while the ingest one is the actual runtime/infra that heavily depends on core. I would like to keep this distinction one way or another, maybe there's other way to enforce that we don't bloat with es dependencies these few classes like IngestDocument, Pipeline and the processors. Ideas?